### PR TITLE
[SFEQS-1131] Add env for io-sign backend

### PR DESCRIPTION
**Pull request by grausof**
This PR adds the environment variables needed to integrate `io-sign` on `io-backend`.

### Type of changes

- [ ] Add new resources
- [X] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [X] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [X] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by gunzip — CHANGES_REQUESTED** [review]

File: src/core/apim_io_backend_v1.tf [hunk 1]
```diff
@@ -281,6 +281,41 @@ module "apim_io_backend_cgn_api_v1" {
...
+## IO_SIGN
+resource "azurerm_api_management_api_version_set" "io_backend_io_sign_api" {
+  name                = format("%s-io-backend-io-sign-api", var.env_short)
+  resource_group_name = module.apim.resource_group_name
+  api_management_name = module.apim.name
+  display_name        = "${local.apim_io_backend_api.display_name} - sign"
+  versioning_scheme   = "Segment"
+}
+
+module "apim_io_backend_io_sign_api_v1" {
```
[the remark this answers is not in the document]
Comment (grausof) [reply src/core/apim_io_backend_v1.tf]: removed!

**Review by pasqualedevita — APPROVED** [review]

**Comment by gunzip** [comment]
@pasqualedevita I've some question about this PR.

1. Is storing the hostkey in the keyvault a requirement for each function deployment?
1. Is this procedure described anywhere?
1. Is communication with devops required in order to enter the function hostkey into the keyvault?
   (I'm asking because, afaik, product teams are not permitted to access the keyvault)
1. Is it possible for the target keyvault to be the one owned by the product team (rather than the "central" one)?

Thank you for your time!! ❤️

**Comment by pasqualedevita** [comment]
> @pasqualedevita I've some question about this PR.
> 
> 1. Is storing the hostkey in the keyvault a requirement for each function deployment?
> 2. Is this procedure described anywhere?
> 3. Is communication with devops required in order to enter the function hostkey into the keyvault?
>    (I'm asking because, afaik, product teams are not permitted to access the keyvault)
> 4. Is it possible for the target keyvault to be the one owned by the product team (rather than the "central" one)?
> 
> Thank you for your time!! ❤️

1. only for function app that are called from a third service like app backend or api management
2. no, but I created a new page to track it https://pagopa.atlassian.net/wiki/spaces/DEVOPS/pages/589103372/Azure+-+Function+App
3. product teams should manage the key vault but I want to check it
4. yes

**Comment by gunzip** [comment]
In order to provide context, I include @manuelcoppotelli in this discussion. 

According to what @pasqualedevita just wrote, we can use "Firma con IO" keyvault (so that we don't have to ask @pasqualedevita every time to insert values) ensuring that the function hostkeys are available to appbackend (for io-func-sign-user) and APIM (for io-func-sign-issuer).

I suppose that the necessity to store function keys into keyvaults is probably due to the fact that the caller service owners (i.e., app-backend or API administration) are not permitted to "inspect" the function service (??) through terraform, so the kayvault is used as a resource to share hostkeys. In a scenario where the owners are able to retrieve functions host keys using Terraform data directives, the keyvault would be unnecessary (isn't it?).

I would investigate whether this setup is feasible using managed identities instead, allowing callers to be able to extract functions hostkeys.

**Pull request merged**